panther_analysis_tool/main.py

```diff
@@ -1106,124 +1122,84 @@ def _run_tests(
-                        continue
-
-                    if aux_function_result["failed"]:
-                        # Mark the test as failed if an auxiliary function fails
-                        test_result[function_name] = test_result["outcome"] = "FAIL"
-                        failed_tests[analysis_id].append(f"{unit_test['Name']}:{function_name}")
+        # print results
+        spec = TestSpecification(
+            id=unit_test["Name"],
+            name=unit_test["Name"],
+            data=unit_test.get("Resource") or unit_test["Log"],
+            mocks=unit_test.get("Mocks", {}),
+            expectations=TestExpectations(detection=unit_test["ExpectedResult"]),
+        )
 
-                    auxiliary_functions_result_message += f"\t\t[{test_result[function_name]}] "
-                    if aux_function_result["has_invalid_type"]:
-                        auxiliary_functions_result_message += "[INVALID TYPE] "
-                    auxiliary_functions_result_message += (
-                        f"[{function_name}] {aux_function_result['display']}\n"
-                    )
-                auxiliary_functions_result_message = auxiliary_functions_result_message.rstrip()
+        test_result = TestCaseEvaluator(spec, result).interpret(
+            ignore_exception_types=ignore_exception_types
+        )
 
-        # print results
-        print("\t[{}] {}".format(test_result["outcome"], unit_test["Name"]))
-        if auxiliary_functions_result_message:
-            print(auxiliary_functions_result_message)
+        _print_test_result(detection, test_result, failed_tests)
 
     return failed_tests
 
 
-def _evaluate_auxiliary_function_result(
-    function_name: str, detection_result: DetectionResult, strict_check: bool
-) -> Dict[str, Any]:
-    """Determine whether an auxiliary function for a rule raised an error"""
-    function_error = getattr(detection_result, f"{function_name}_exception")
-    output = getattr(detection_result, f"{function_name}_output")
-    is_defined = getattr(detection_result, f"{function_name}_defined")
-    failed = function_error is not None
-
-    # For backwards compatibility we can accept invalid destination names,
-    # as long as a string is returned. Strict check is enabled when users
-    # pass destination names explicitly through command-line parameters.
-    if function_name == "destinations" and failed and not strict_check:
-        # Otherwise we fall back to a best-effort check of the return type only
-        if isinstance(function_error, UnknownDestinationError):
-            exc: UnknownDestinationError = function_error
-            failed = not _check_destinations_type(exc.result())
-
-    # The function has not been executed
-    if not is_defined:
-        display = None
+def _print_test_result(
+    detection: Detection, test_result: TestResult, failed_tests: DefaultDict[str, list]
+) -> None:
+    status_pass = "PASS"  # nosec
```

Comment:
Out of curiosity, why the `# nosec` directive?

Reply:
This is for a [bandit](https://pypi.org/project/bandit/) check:

```
Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'PASS'
   Severity: Low   Confidence: Medium
   Location: panther_analysis_tool/main.py:1164
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
1163    ) -> None:
1164        status_pass = "PASS"
1165        status_fail = "FAIL"
```